### ISSUE TITLE
Viewing of Profiles

### ISSUE THREAD

**Issue by jlong5795** [benawad/vsinder#155]
I selected that I am a man who is looking for love and wants to see men. Why am I given women and straight men?

**Comment by unDeadHerbs** [comment]
I expect this is because people are lying on their profiles, as in #117.  But if you want to be sure, you post some of the profiles for @benawad to check against the database that the filters are working.  Please report the flagrant profiles.

**Comment by pawan-maurya-dev** [comment]
@benawad don't you think the gender selection should be visible for the matches. This'll help in reporting the culprits.

**Comment by unDeadHerbs** [comment]
The problem is that gender is a personal opinion question.  A user can't be known to be lying, only suspected.  And, while the app could ask for the user's sex instead, this probably wouldn't fix the issue and would be rude.

I see two different options that might resolve this:
1. Offer more diverse gender selection and filters in the hopes of confounding people into telling the truth.
2. Re-word the gender filter questions to decrease lying. 

I have some ideas on option 1, but they are rather long winded and much work to implement.  For option 2, renaming the "show me code from:" to "Homosexual", "Heterosexual", "Non-Binary" would make cishet males feel more emasculated to lie.  (I suspect cishets are the majority of the user base on this platform, as they usually are.)  This isn't an ideal solution, but it might work.

**Comment by pawan-maurya-dev** [comment]
@unDeadHerbs @benawad if you want to appropriate the gender identity preference for people, it's impossible to solve this issue with gender filters.

e.g. a He might identifies as she.

**Comment by unDeadHerbs** [comment]
I agree, the problem is the number of people apparently willfully filling in things other than what they identify as.

**Comment by pawan-maurya-dev** [comment]
Then what is the criteria for github profile ban once somebody reports a profile by just looking at the profile picture.

**Comment by unDeadHerbs** [comment]
I'd believe that's the current solution. It's not good, but it's better than nothing.

**Comment by unDeadHerbs** [comment]
@benawad can we have an estimate of the size of this problem based on the number of reports?

**Comment by benawad** [comment]
I would guesstimate less than 100

**Comment by unDeadHerbs** [comment]
I don't know how many users there are, but that sounds like a small portion.  Hopefully they were a symptom of the small userbase at the start and will weed themselves out by just not getting any matches.

**Comment by unDeadHerbs** [comment]
If it's that minor of a problem, I'm tempted to close this and let reporting do is work.

**Comment by jlong5795** [comment]
I think the best way to address this is to update profiles so that they show the person's gender identity. That way a person can have a bit more information about identity vs appearance.

**Comment by unDeadHerbs** [comment]
That could work well, I've noticed a number of people who list their pronouns in their profile.  This could be made more official by giving the user a text box to fill them in and then display that at the bottom of the card.  This would be a non-functional change, as making any sort of filter on that quickly leads to problems and regex.

Proposed change:
- Add text entry box labeled "pronouns" to "edit profile"
- Default all existing users to "He/Him", "She/Her", and "They/Them" based on existing gender selection
- Add a parenthesized display of this between the user name and age
- Disallowed empty pronouns by resetting to the default

**Comment by nyanpasu64** [comment]
> Proposed change:
> 
>     * Add text entry box labeled "pronouns" to "edit profile"
> 
>     * Default all existing users to "He/Him", "She/Her", and "They/Them" based on existing gender selection
> 
>     * Add a parenthesized display of this between the user name and age
> 
>     * Disallowed empty pronouns by resetting to the default

Please retain an option to leave your pronoun field empty, or possibly add custom pronouns as well.

**Comment by unDeadHerbs** [comment]
That's why I'm suggesting a textbox, it provides plenty of freedom.  In don't think there's any new chance of abuse with this, seeing as it's in the same place as the user profile bio.

**Comment by jlong5795** [comment]
The issue with a textbox is that it leads to the same ambiguity that currently exists. A set of responses locks the user into a declaration.

**Comment by unDeadHerbs** [comment]
I don't meant this as a replacement to the current radial buttons but as an addition to them.  It provides more expressiveness without losing the filtering ability.  But I'm all ears for different ideas.

**Comment by nyanpasu64** [comment]
One possible design:

- keep male/female/non-binary options (possibly rename it to Other)
- make it visible on people's profiles. I've seen people with a male name and picture and "m" in their profile, but showed up when filtering by "female or non-binary". but i don't know which one they picked.
  - this has the disadvantage that if you want to keep your gender private, you can no longer do so.
  - one approach is to make visibility optional, another is to add an "unspecified" gender which is a separate option when users filter by gender
- add an optional text field for gender details and pronouns, also visible on people's profiles
